init.lua

```diff
@@ -37,57 +31,129 @@ local config = {
     window_w = 160, window_h = 220,
 }
 
+---@param s string
+---@return string
+local function sanitize_filename(s)
+    s = tostring(s or '')
+    s = s:gsub('[^%w_%-%s]', ''):gsub('%s+', '_')
+    if s == '' then s = 'Unknown' end
+    return s
+end
+
+---@return string
+local function get_config_path()
+    local toon = sanitize_filename(mq.TLO.Me.CleanName() or 'Unknown')
+    return mq.configDir .. ('/Arrow.%s.lua'):format(toon)
+end
+
 ---@param path string
 ---@return boolean
 local function file_exists(path)
-    local f = io.open(path, "r")
+    local f = io.open(path, "rb")
     if f then f:close() return true end
     return false
 end
 
-local function load_config()
-    if not file_exists(config_file) then return end
-    local ok, ini = pcall(LIP.load, config_file)
-    if ok and ini and ini.Settings then
-        config.arrow_scale = tonumber(ini.Settings.arrow_scale) or config.arrow_scale
-        config.arrow_color = {
-            tonumber(ini.Settings.color_r) or config.arrow_color[1],
-            tonumber(ini.Settings.color_g) or config.arrow_color[2],
-            tonumber(ini.Settings.color_b) or config.arrow_color[3],
-            tonumber(ini.Settings.color_a) or config.arrow_color[4],
-        }
-        config.window_x = tonumber(ini.Settings.window_x) or config.window_x
-        config.window_y = tonumber(ini.Settings.window_y) or config.window_y
-        config.window_w = tonumber(ini.Settings.window_w) or config.window_w
-        config.window_h = tonumber(ini.Settings.window_h) or config.window_h
+---@param v any
+---@param indent string|nil
+---@return string
+local function serialize_lua(v, indent)

```

Comment:
Pick is provided by mq. you shouldn't write your own

https://docs.macroquest.org/lua/pickle/